File: golabs/gocloud/godata/go-gumball-riak/gumball-riak/src/github.com/basho/taste-of-riak/nodejs/Ch03-Msgy-Schema/repositories/timeline-repository.js
'use strict';

var Repository = require('./repository');
var Timeline = require('../models/timeline');
var inherits = require('util').inherits;

function TimelineRepository(client) {
    Repository.call(this, client);
    this.bucketName = 'Timelines';
}

inherits(TimelineRepository, Repository);

TimelineRepository.prototype.buildModel = function (rslt) {
    if (rslt.values) {
        var rv = null,
            riakObj = rslt.values.shift();
        if (riakObj && riakObj.value) {
            var timelineJsonObj = riakObj.value;
            var model = new Timeline(timelineJsonObj.data.id);
            timelineJsonObj.data.msgKeys.forEach(function (msgKey) {
                model.addMsg(msgKey);
            });
            rv = model;
        }
        return rv;
    }
};

module.exports = TimelineRepository;

File: golabs/godata/go-gumball-riak/gumball-riak/src/github.com/basho/taste-of-riak/csharp/Ch03-Msgy-Schema/Repositories/TimelineRepository.cs
namespace Ch03_Msgy_Schema.Repositories
{
    using Models;
    using RiakClient;

    public class TimelineRepository : Repository<Timeline>
    {
        const string BUCKET_NAME = "Timelines";

        public TimelineRepository(IRiakClient client)
            : base(client)
        {
        }

        protected override string BucketName
        {
            get { return BUCKET_NAME; }
        }
    }
}
